src/Connection.cpp

```diff
@@ -178,6 +178,29 @@ uint64_t Connection::getMethodCallTimeout() const
     return timeout;
 }
 
+Slot Connection::addMatch(const std::string& match, message_handler callback)
+{
+    auto matchInfo = std::make_unique<MatchInfo>(MatchInfo{std::move(callback), *this, {}});
+
+    auto messageHandler = [](sd_bus_message *sdbusMessage, void *userData, sd_bus_error */*retError*/) -> int
+    {
+        auto* matchInfo = static_cast<MatchInfo*>(userData);
+        auto message = Message::Factory::create<PlainMessage>(sdbusMessage, &matchInfo->connection.getSdBusInterface());
+        matchInfo->callback(message);
+        return 0;
+    };
+
+    auto r = iface_->sd_bus_add_match(bus_.get(), &matchInfo->slot, match.c_str(), std::move(messageHandler), matchInfo.get());
+    SDBUS_THROW_ERROR_IF(r < 0, "Failed to add match", -r);
+
+    return {matchInfo.release(), [this](void *ptr)
+    {
+        auto* matchInfo = static_cast<MatchInfo*>(ptr);
+        iface_->sd_bus_slot_unref(matchInfo->slot);
+        std::default_delete<MatchInfo>{}(matchInfo);
+    }};
+}
+
```

Comment:
@dleeds-cpi Yes that was left TBD for the weekend :) Pushed already. What sd-bus also provides is an async way to add match rules... This I didn't implement, I guess it's not so super useful, what do you think?

Reply:
It could be useful if you're registering 10s or 100s of match rules and want to avoid blocking the calling thread. However, I would prioritize adding async property get/set support before adding async match rule support.